src/helpers/figures/charts/abstract_chart.ts

```diff
@@ -21,11 +21,15 @@ import { Validator } from "../../../types/validator";
 export abstract class AbstractChart {
   readonly sheetId: UID;
   readonly title: string;
+  readonly isTitleCellReferenced?: boolean;
+  readonly titleCellReference?: string;
   abstract readonly type: ChartType;
   protected readonly getters: CoreGetters;
 
   constructor(definition: ChartDefinition, sheetId: UID, getters: CoreGetters) {
     this.title = definition.title;
+    this.isTitleCellReferenced = definition.isTitleCellReferenced || false;
+    this.titleCellReference = definition.titleCellReference || "";
```

Comment:
In the definition I think it's better to have a single key for the title (see my previous comment), but here you don't haev a choice to have both. But `titleCellReference` should be a `Range`, and not a string. And it should be transormes like the other ranges of the chart. 

Currently you title reference doen't react at all to the addition/removal of columns rows, neither in the `updateRanges` nor in the collaborative transforms `transformDefinition`.

You're missing a lot of tests here.